### PR TITLE
L54: Python Server Wait API

### PR DESCRIPTION
**Pull request by lidizheng**
gRFC: https://github.com/lidizheng/proposal/blob/python-wait/L54-python-server-wait.md
Group Thread: https://groups.google.com/forum/#!topic/grpc-io/GhJl_JHfaHA

### CODE REVIEW

File: L54-python-server-wait.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/GhJl_JHfaHA
+
+## Abstract
+
+Add a `wait_for_termination` API for Python server that can block in any thread
+until the server is stopped or terminated.
+
+## Background
+
+Currently, the gRPC Python server only provides a `server.Start()` API, but
```
Comment (gnossen) [inline L54-python-server-wait.md]: `s/Start/start/`

File: L54-python-server-wait.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+in Java as `blockUntilShutdown()`, in Golang as `Serve`. This gRFC is
+introducing the Python version of it.
+
+### Related Proposals: 
+
+None.
+
+## Proposal
+
+This gRFC propose to add following API to `grpc.Server` interface:
```
Comment (gnossen) [inline L54-python-server-wait.md]: `s/propose to add/proposes that we add/`

File: L54-python-server-wait.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        
+        1) Calling `stop` on the server in another thread;
+        2) The `__del__` of the server object is invoked.
+        """
+        raise NotImplementedError()
+```
+
+## Rationale
+
+### Alternative Design: Await For Signal
```
Comment (gnossen) [inline L54-python-server-wait.md]: I definitely think the way you chose is the way to go. The application author should be in charge of what runs on which thread. They can write their own signal handler to run on the main thread and call `server.stop()`.

Comment (lidizheng) [reply L54-python-server-wait.md]: Glad to see that we have similar view of how this API should be done :)
I added another alternative design section describing the idea proposed by @mehrdada in the implementation PR.

**Pull request merged**